packages/core/src/components/tree/treeNode.tsx

```diff
@@ -144,11 +151,17 @@ export class TreeNode<T = {}> extends React.Component<ITreeNodeProps<T>, {}> {
 
     private handleCaretClick = (e: React.MouseEvent<HTMLElement>) => {
         e.stopPropagation();
-        const { isExpanded, onCollapse, onExpand } = this.props;
+        const { disabled, isExpanded, onCollapse, onExpand } = this.props;
+        if (disabled) {
+            return;
```

Comment:
the `e.stopPropagation()` in the disabled case feels a little fishy... can we try a different approach where we don't attach the event handlers in `render` if `disabled === true` (we prefer the explicit boolean check instead of coercing a potentially undefined value to boolean)? that would reduce the code diff here as well